### PR TITLE
Bug 1156161 - Add manifestHash and packageHash fields to preload apps' registry in webapps.json

### PR DESCRIPTION
**Pull request by rickychien**

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=6c17d9a43df4bdad6952416fd6fcdae3750bc29f

File: build/utils-xpc.js [hunk 1]
```diff
@@ -310,7 +310,33 @@ function getUUIDMapping(config) {
   }
   return UUID_MAPPING;
 }
-exports.getUUIDMapping = getUUIDMapping;
+
+function getMD5hash(filePath) {
+  var file = Cc['@mozilla.org/file/local;1']
+    .createInstance(Ci.nsILocalFile);
```
Comment (fabricedesre) [inline build/utils-xpc.js]: nit: use nsIFile instead of nsILocalFile

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4e11043cf1f946f76bcf5e12baee53093f2b096b

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3ec627b865fd3d10bc6190ad9a40948f975061a5

**Comment by cctuan** [comment]
test fail? 
Exception: Error: -*- build/utils.js: [Exception... "Component returned failure code: 0x80520012 (NS_ERROR_FILE_NOT_FOUND) [nsIFileInputStream.init]" nsresult: "0x80520012 (NS_ERROR_FILE_NOT_FOUND)" location: "JS frame :: resource://gre/modules/commonjs/toolkit/loader.js -> file:///home/tester/git_checkout/build/utils-xpc.js :: getFileContent :: line 92" data: no]
make[1]: *** [build-app] Error 1
make[1]: *** [build-app] Error 1
TEST-UNEXPECTED-FAIL | build/test/integration/core_app_location_test.js | Make with desktop custom core apps location make with desktop custom core apps location
AssertionError: expected [Error: Command failed: 2015-04-24 03:17:23: stackwalker.cc:125: INFO: Couldn't load symbols for: /home/tester/git_checkout/b2g_sdk/39.0a1-2015-03-05-16-02-02/b2g/libxul.so|11E2B0D603E2C867B31E7443EA59FD7F0
make[1]: *** [build-app] Error 1
make[1]: *** [build-app] Error 1
make: *** [build-test-integration] Error 1

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=fbdfd839f09cd02c62497dad77e3b0b86fe7454e

**Pull request merged**